internal/provider/models/destinations/loki.go

```diff
@@ -159,7 +165,11 @@ func LokiDestinationToModel(plan *LokiDestinationModel, component *Destination)
 				value := obj["label_value"].(string)
 				labelMap[key] = value
 			}
-			plan.Labels = basetypes.NewMapValueMust(plan.Labels.ElementType(nil), MapAnyToMapValues(labelMap))
+			labelType := plan.Labels.ElementType(context.TODO())
```

Comment:
TIL, what's `context.TODO`, and why this versus the `Background()` one ?

Reply:
ElementType does not use the context it is passed.
I do have go static checker which advises to pass a TODO context in such cases: https://staticcheck.io/docs/checks#SA1012

Reply:
ah nice, thanks!

Reply:
I found [this response](https://www.mongodb.com/community/forums/t/go-why-context-todo-and-not-context-background/5009/3) regarding the use in MongoDB drivers helpful:

> As you mentioned in your response, they are the same context (no timeout/deadline and no associated values). The difference is semantic. I think of context.Background as “I’m deliberately passing in an empty context” and context.TODO as “there should be some other context here, which could be empty, but I’m not sure what the right value is yet so here’s a placeholder”.
>
> We use context.TODO() in the examples embedded in our documentation because the correct value would depend on the calling function, which is usually code from the user’s application. In this case, it’s up to the user to decide what the context should be, which usually depends on things like the maximum time they want the operation to take and whether or not the calling function has a context that it can propagate.

Since we're sure that we need a nil context at this code based on authoring the provider, we should probably sick with `context.Background()` every time we need a nil context object. Using `TODO()` makes it seem like we're not sure or that the implementation is incomplete.